### PR TITLE
NFC Code cleanup to core task class

### PR DESCRIPTION
**Pull request by mattwire**
Overview
----------------------------------------
Just commenting and variable renaming towards making this more generic.

Identified during investigation for dev/core#158

Technical Details
----------------------------------------
CRM_Core_Form_Task::preProcessCommon() is currently only used by CRM_Case_Form_Task.

### CODE REVIEW

File: CRM/Case/Form/Task.php [hunk 1]
```diff
@@ -40,6 +40,13 @@ class CRM_Case_Form_Task extends CRM_Core_Form_Task {
   // Must be set to entity shortname (eg. event)
   static $entityShortname = 'case';
 
+  /**
+   * Must be set to queryMode
+   *
+   * @var int
+   */
+  static $queryMode = CRM_Contact_BAO_Query::MODE_CASE;
```
Comment (colemanw) [inline CRM/Case/Form/Task.php]: Where is this used?

Comment (eileenmcnaughton) [reply CRM/Case/Form/Task.php]: I couldn't find any instances?

**Comment by eileenmcnaughton** [comment]
Just a testing note - this messes with the $useTable variable so we need to check that it is still possible to export more than 500 contacts - eg from manage groups and normal searches only retrieve the first 500

File: CRM/Case/Form/Task.php (hunk 1 above)
Comment (mattwire) [reply CRM/Case/Form/Task.php]: https://github.com/civicrm/civicrm-core/pull/12316/files#diff-8e58e0ef24ffee40975a92a797f7acecR135

It's a baby step towards making that preProcessCommon function generic enough to handle more than just the case task.

**Comment by eileenmcnaughton** [comment]
@mattwire I worked through this refactor locally & it makes sense. It changes 'queryMode' to be more correct - does that fix any bug? And if so is there something still needing backporting to the rc?

I'm merging this but can we follow up by adding a getter on queryMode ?

```
$form::getQueryMode();
```
is, IMHO, less ugly than
```
$form::$queryMode
```

**Pull request merged**